src/matrix_validator/validator.py

```diff
@@ -10,6 +10,9 @@
 
 logger = logging.getLogger(__name__)
 
+CURIE_REGEX = r"^[A-Za-z_\.]+:.+$"
```

Comment:
I think this is too permissive, especially the local ID part.

See also https://github.com/biopragmatics/curies

```
NCNAME_PATTERN = r"[A-Za-z_][A-Za-z0-9\.\-_]*"
"""A regex for prefixes, from https://www.w3.org/TR/1999/REC-xml-names-19990114/#NT-NCName.

LOCAL_UNIQUE_IDENTIFIER_PATTERN = r"(/[^\s/][^\s]*|[^\s/][^\s]*|[^\s]?)"
"""A regex for local unique identifiers in CURIEs, based on https://www.ietf.org/rfc/rfc3987.txt
```